dbos/_recovery.py

```diff
@@ -4,22 +4,50 @@
 import traceback
 from typing import TYPE_CHECKING, Any, List
 
+from sqlalchemy.exc import DatabaseError
+
 from ._core import execute_workflow_by_id
 from ._error import DBOSWorkflowFunctionNotFoundError
+from ._sys_db import GetPendingWorkflowsOutput
 
 if TYPE_CHECKING:
     from ._dbos import DBOS, WorkflowHandle
 
 
-def startup_recovery_thread(dbos: "DBOS", workflow_ids: List[str]) -> None:
+def clear_pending_workflow_queue_assignement(
+    dbos: "DBOS", workflow: GetPendingWorkflowsOutput
+) -> None:
+    dbos.logger.debug(
+        f"Clearing queue assignment for workflow: {workflow.workflow_uuid}"
+    )
+    try:
+        dbos._sys_db.clear_queue_assignment(workflow.workflow_uuid)
+    except DatabaseError as e:
+        if getattr(e.orig, "pgcode", None) == "40001":
+            dbos.logger.debug(
+                f"Workflow {workflow.workflow_uuid} queue assignment is already being cleared"
+            )
+        else:
+            raise e
```

Comment:
If this raises, there is a more serious issue. But I actually don't know that we want to throw the entire process for it. Today the risk is that this goes un-noticed. Tomorrow hopefully the UI will surface workflows that failed to recover?


cc @kraftp